llpc/context/llpcPipelineContext.cpp

```diff
@@ -374,6 +379,12 @@ void PipelineContext::setOptionsInPipeline(Pipeline *pipeline, Util::MetroHash64
     else
       shaderOptions.vgprLimit = VgprLimit;
 
+    // Enable waterfall load scalarization when vgpr limit is set.
```

Comment:
Unless there is some existing convention that you are following here, I would suggest:
```
shaderOptions.scalarizeWaterfallLoads = ScalarizeWaterfallDescriptorLoads.getNumOccurrences() ? ScalarizeWaterfallDescriptorLoads : (some other calculation here);
```
so that the command line option can always be used to force this either on or off.